User management rework

nika-gelashvili opened the pull request:
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | yes/no
| Fixed issues  | comma-separated list of tickets # fixed by the PR, if any

nika-gelashvili committed:
WIP: Implement Employee management.

nika-gelashvili committed:
Implement search model form user department.
Implement modal in EmployeeList.vue for editing user data and assigning department and role.
Assign data to columns in EmployeeList.vue table accordingly.

nika-gelashvili committed:
Create user_roles table with according model and resource.
Add endpoints for updating user data and getting data for dropdowns.
Add multiselect node module.

nika-gelashvili committed:
Fix getModalDropdownData function not being in mapActions.

nika-gelashvili committed:
Fix data for dropdowns.
Add hide modal in EmployeeListForm.vue.
Add url for getting dropdown data.

nika-gelashvili committed:
Fix employees and invitations nav bar active class.
Change invitations table visual.

nika-gelashvili committed:
Remove country_id from user_departments table.
Edit user department model and resource to reflect change.

nika-gelashvili committed:
Change UserDepartmentSearch.php to get country data from department.
Add department extra field to CountryResource.php

nika-gelashvili committed:
Create PositionModel and RoleModel for user position and roles.

nika-gelashvili committed:
Change name of EmployeeListForm.vue to EmployeeFormModal.vue.
Change name of EmployeeListFormModel.js to EmployeeModel.js.
Change logic for getting dropdown data, remove getting country and department data from actionGetDropdown and use country endpoint instead(needs to be changed).

nika-gelashvili committed:
Edit card styles remove labels form each role and permission added make labels for cards move buttons to right, minimize role remove button.
Remove vue-multiselect and using b-form-select for departments.
Use dropdown for countries instead of input with options.
Implement model usage for roles and positions.

nika-gelashvili committed:
Add byUserId search to UserDepartmentQuery.php.
Add buId search to UserQuery.php.
Add byUserId search to UserRoleQuery.php.
Add authManager to web.php config.

nika-gelashvili committed:
Add action for updating user data(user, user profile, user department, user roles).
Add function updateRoles to UserResource.php for creating updating and deleting user roles.
Add function updateUserDepartments for creating updating and deleting user departments.
Add new field roles in UserResource.php for user roles.
Add relation to department table in User.php.
Add function get roles for roles files in UserResource.php.

nika-gelashvili committed:
Rename PositionModel.js to UserDepartmentModel.js and changing it by adding id and department_id attributes, removing departments array attribute and changing attribute name to position.

nika-gelashvili committed:
Add updateUserData service for endpoint actionUpdateUserData.
Fix country names not displaying in table.

nika-gelashvili committed:
Merge showModal and modalEmployee states in employees.js into modal.
Change showEmployeeModal mutation to work with above change and save data using deconstruction.
Change hideModal mutation to reflect states change.

nika-gelashvili committed:
Improve EmployeeModel.js constructor.
Rename position into userDepartments.
Add success logic to submission.

nika-gelashvili committed:
Improve employee form modal visually

nika-gelashvili committed:
Implement InputWidget.vue for dropdowns in EmployeeFormModal.vue.
Remove console.log use for testing in EmployeeModel.js.
Make fields respective model arrays in EmployeeModel.js.
Add rules and labels to fields in RoleModel.js nad UserDepartmentModel.js.
Comment delete in toJSON in UserDepartmentModel.js as it was removing necessary data when initialising model.

nika-gelashvili committed:
Fix ul styles in EmployeeList.vue

nika-gelashvili committed:
Change input-widget's value-field and text-field default values.
Edit EmployeeFormModal.vue input-widget usage accordingly.

nika-gelashvili committed:
Change setMultipleErrors in BaseModel.js as several formats of error messages are returned from backend.
Add check for data format returned from backend and set error messages accordingly.
Add logic for submit response error handling in EmployeeFormModal.vue.

nika-gelashvili committed:
Merge branch 'master' into user-management-rework

# Conflicts:
#	models/User.php
#	models/query/UserQuery.php
#	vue/package.json
#	vue/src/core/components/input-widget/BaseModel.js
#	vue/src/core/components/layout/DefaultLayout/DefaultLayout.vue
#	vue/src/shared/router.js

nika-gelashvili committed:
Remove ContactForm.php as it was not used.
Fix yii\rest\UrlRule not being imported but used in web.php.
Implement verbs filter in EmployeeController.php.

nika-gelashvili committed:
Move employees.js and invitations.js to vuex modules and separate mutations, actions and states.
Remove old invitations.js, employees.js and user.js as it wont be used anymore.
Change createNamespacedHelpers parameter according to changes described above.

nika-gelashvili committed:
Change UserController.php name to AuthController.php and move to users module.
Move SiteController.php to setup module and change errorAction in config web.php to SiteController.php's new position.
Change authService.js login, register, resetPasswordLink, passwordReset and checkToken functions to use AuthController actions accordingly.

nika-gelashvili committed:
Change MyUserController.php name to UserController.php.
Move UserController.php to users module.
Change actions.js service path's to use new UserController.php path.

nika-gelashvili committed:
Move User, LoginForm.php, SignupForm.php models to users module.
Move UserQuery.php to users module.
Change usages of models and query accordingly.

nika-gelashvili committed:
Add auth controller to urlManager rules.
Change my-user to user in  urlManager rules.

nika-gelashvili committed:
Remove user_role table migration with model, query and resource.
Move role constants to UserResource.php and add static function for getting roles.
Add first name and last name in UserResource.php fields.

nika-gelashvili committed:
Remove UserProfile.php as it is not needed any more as all profile data is in users table.
Remove relation from User model.
Change getDisplayName function accordingly.
Change SignupForm.php function signup accordingly.
Remove userProfile from employeesService.js expand parameter.
Edit employee mutations.js to use user models first and last name.
Remove user profile save from actionUpdateUserData.
Change actionGetDropdown to use roles from UserResource.

thecodeholic commented on modules/v1/users/controllers/EmployeeController.php:
```diff
@@ -0,0 +1,120 @@
+
+class EmployeeController extends ActiveController
+{
+    public $modelClass = UserDepartmentResource::class;
```
We should use UserResource here

thecodeholic commented on modules/v1/users/controllers/EmployeeController.php:
```diff
@@ -0,0 +1,120 @@
+    public function actions()
+    {
+        $actions = parent::actions();
+        unset($actions['update'], $actions['view'], $actions['index']);
```
Why to remove native actions. We can use them...

thecodeholic commented on modules/v1/users/controllers/EmployeeController.php:
```diff
@@ -0,0 +1,120 @@
+    protected function verbs()
+    {
+        $verbs = parent::verbs();
+        $verbs['index'] = ['GET', 'HEAD', 'OPTIONS'];
```
We don't need to touch index and update actions at all.

thecodeholic commented on modules/v1/users/models/search/UserDepartmentSearch.php:
```diff
@@ -0,0 +1,52 @@
+use yii\base\Model;
+use yii\data\ActiveDataProvider;
+
+/**
```
Move this comment up immediatelly after php opening tag

thecodeholic commented on modules/v1/users/resources/UserDepartmentResource.php:
```diff
@@ -0,0 +1,41 @@
+    {
+        return [
+            'id',
+            'position',
```
It will be also good to return user_id and department_id here.

thecodeholic commented on modules/v1/users/resources/UserResource.php:
```diff
@@ -42,14 +54,116 @@ public function fields()
+    public static function getUserRoles() {
+        return [
+            [
+                'value' => 'user',
```
This should not be like this.
We should have value to be constant and text must be user text written in Yii:t( method.

thecodeholic commented on vue/src/modules/User/Employees/EmployeeModel.js:
```diff
@@ -0,0 +1,65 @@
+
+  rules = {
+    first_name: [
+      {rule: 'required'},
```
You can write string `required` instead of object.

thecodeholic commented on vue/src/modules/User/Employees/EmployeeModel.js:
```diff
@@ -0,0 +1,65 @@
+    last_name: i18n.t('Last Name'),
+  };
+
+  toJSON() {
```
This method is useless.

thecodeholic commented on vue/src/modules/User/Employees/employeesService.js:
```diff
@@ -0,0 +1,20 @@
+
+  get(params = {
+    sort: '-created_at',
+    expand: 'userDepartments, userDepartments.department, userDepartments.department.country'
```
Remove white spaces after comma

nika-gelashvili committed:
Resolve comments

nika-gelashvili committed:
Remove index action prepareDataProvider override.
Remove prepareDataProvider function as it is not necessary.
Remove actionUpdate and unsetting default update action.

nika-gelashvili committed:
Add 2 extra fields roles and userDepartmentsData.
Override rules to add extra field rules.
Override model save and implement updateUserDepartments and updateRoles functions into save.

nika-gelashvili committed:
Change model in relation from Department to DepartmentResource.

nika-gelashvili committed:
Implement blamable and timestamp behaviour.

nika-gelashvili committed:
Fix updateUserData url string.

nika-gelashvili committed:
Implement logic to change model parameter from userDepartments to userDepartmentsData and send data this way as userDepartments can't be used as model field, because relation field has the same name.

nika-gelashvili committed:
Implement authorisation data migration.

thecodeholic committed:
Remove unnecessary methods from UserDepartmentSearch

thecodeholic committed:
Delete migration of user_profiles table

thecodeholic committed:
Make UserDepartment model labels translatale

thecodeholic committed:
Minor improvments

thecodeholic committed:
Merge branch 'master' into user-management-rework

# Conflicts:
#	migrations/m200918_124646_create_user_profiles_table.php
#	vue/package.json
